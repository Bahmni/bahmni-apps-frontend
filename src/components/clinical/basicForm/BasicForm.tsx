import React, { useEffect } from 'react';
import {
  Dropdown,
  DatePicker,
  DatePickerInput,
  Grid,
  Column,
  MenuItemDivider,
  Loading,
} from '@carbon/react';
import { useTranslation } from 'react-i18next';
import { useLocations } from '@hooks/useLocations';
import { useEncounterConcepts } from '@hooks/useEncounterConcepts';
import { useActivePractitioner } from '@hooks/useActivePractitioner';
import { useEncounterDetailsStore } from '@stores/encounterDetailsStore';
import { formatDate } from '@utils/date';
import * as styles from './styles/BasicForm.module.scss';
import { FhirEncounter } from '@types/encounter';

/**
 * BasicForm props
 * @interface BasicFormProps
 * @property {FhirEncounter} activeVisit - The active visit for deriving visit type
 */
interface BasicFormProps {
  activeVisit: FhirEncounter;
}
const BasicForm: React.FC<BasicFormProps> = ({ activeVisit }) => {
  const { t } = useTranslation();
  const { locations, loading: loadingLocations } = useLocations();
  const { encounterConcepts, loading: loadingEncounterConcepts } =
    useEncounterConcepts();
  const { practitioner, loading: loadingPractitioner } =
    useActivePractitioner();

  const {
    selectedLocation,
    selectedEncounterType,
    selectedVisitType,
    encounterParticipants,
    consultationDate,
    isEncounterDetailsFormReady,
    setSelectedLocation,
    setSelectedEncounterType,
    setSelectedVisitType,
    setEncounterParticipants,
    setEncounterDetailsFormReady,
  } = useEncounterDetailsStore();

  useEffect(() => {
    if (locations.length > 0 && !selectedLocation) {
      setSelectedLocation(locations[0]);
    }
  }, [locations, selectedLocation]);

  useEffect(() => {
    if (encounterConcepts && !selectedEncounterType) {
      const consultationType = encounterConcepts.encounterTypes.find(
        (item) => item.name === 'Consultation',
      );
      if (consultationType) {
        setSelectedEncounterType(consultationType);
      }
    }
  }, [encounterConcepts, selectedEncounterType]);

  useEffect(() => {
    if (encounterConcepts && activeVisit && !selectedVisitType) {
      const activeVisitId = activeVisit.type?.[0]?.coding?.[0]?.code || '';
      const visitType = encounterConcepts.visitTypes.find(
        (item) => item.uuid === activeVisitId,
      );
      if (visitType) {
        setSelectedVisitType(visitType);
      }
    }
  }, [encounterConcepts, activeVisit, selectedVisitType]);

  useEffect(() => {
    if (practitioner && encounterParticipants.length === 0) {
      setEncounterParticipants([practitioner]);
    }
  }, [practitioner, encounterParticipants.length]);

  // Set form ready state based on loading states
  useEffect(() => {
    const isAllDataLoaded =
      !loadingLocations && !loadingEncounterConcepts && !loadingPractitioner;
    setEncounterDetailsFormReady(isAllDataLoaded);
  }, [loadingLocations, loadingEncounterConcepts, loadingPractitioner]);

  // TODO used the current practitioner as the only available option
  // Later this will be replaced with multiple practitioners
  const availablePractitioners = practitioner ? [practitioner] : [];

  const formattedDate = formatDate(consultationDate);

  if (!isEncounterDetailsFormReady) {
    return (
      <Grid condensed={false} narrow={false}>
        <Column sm={4} md={8} lg={16}>
          <Loading description={t('LOADING_FORM_DATA')} withOverlay={false} />
        </Column>
      </Grid>
    );
  }

  return (
    <>
      <Grid condensed={false} narrow={false}>
        <Column sm={4} md={8} lg={5} xl={12} className={styles.column}>
          <Dropdown
            id="location-dropdown"
            titleText={t('LOCATION')}
            label={t('SELECT_LOCATION')}
            items={locations}
            itemToString={(item) => (item ? item.display : '')}
            initialSelectedItem={selectedLocation}
            onChange={({ selectedItem }) => setSelectedLocation(selectedItem)}
            disabled
            size="md"
            type="default"
          />
        </Column>
        <Column sm={4} md={8} lg={5} xl={12} className={styles.column}>
          <Dropdown
            id="encounter-type-dropdown"
            titleText={t('ENCOUNTER_TYPE')}
            label={t('SELECT_ENCOUNTER_TYPE')}
            items={encounterConcepts?.encounterTypes || []}
            itemToString={(item) => (item ? item.name : '')}
            initialSelectedItem={selectedEncounterType}
            onChange={({ selectedItem }) =>
              setSelectedEncounterType(selectedItem)
            }
            disabled
            size="md"
            type="default"
          />
        </Column>
        <Column sm={4} md={8} lg={5} xl={12} className={styles.column}>
          <Dropdown
            id="visit-type-dropdown"
            titleText={t('VISIT_TYPE')}
            label={t('SELECT_VISIT_TYPE')}
            items={encounterConcepts?.visitTypes || []}
            itemToString={(item) => (item ? item.name : '')}
            initialSelectedItem={selectedVisitType}
            onChange={({ selectedItem }) => setSelectedVisitType(selectedItem)}
            disabled
            size="md"
            type="default"
          />
        </Column>
        <Column sm={4} md={8} lg={5} className={styles.column}>
          <Dropdown
            id="practitioner-dropdown"
            titleText={t('PARTICIPANTS')}
            label={t('SELECT_PARTICIPANTS')}
            items={availablePractitioners}
            itemToString={(item) =>
              item?.person?.preferredName?.display
                ? item.person.preferredName.display
                : ''
            }
            initialSelectedItem={practitioner} // Show the selected participant
            onChange={({ selectedItem }) => {
              // For now, single select behavior
              if (selectedItem) {
                setEncounterParticipants([selectedItem]);
              }
            }}
            disabled
            size="md"
            type="default"
          />
        </Column>
        <Column sm={4} md={8} lg={5} className={styles.column}>
          <DatePicker datePickerType="single" dateFormat="d/m/Y">
            <DatePickerInput
              id="encounter-date-picker-input"
              placeholder={formattedDate.formattedResult}
              title={t('ENCOUNTER_DATE')}
              labelText={t('ENCOUNTER_DATE')}
              defaultValue={formattedDate.formattedResult}
              disabled
            />
          </DatePicker>
        </Column>
      </Grid>
      <MenuItemDivider />
    </>
  );
};

export default BasicForm;
