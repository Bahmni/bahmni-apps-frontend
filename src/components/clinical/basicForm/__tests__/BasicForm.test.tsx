import React from 'react';
import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import BasicForm from '../BasicForm';
import { useTranslation } from 'react-i18next';
import { Provider } from '@types/provider';

// Mock the hooks
jest.mock('react-i18next', () => ({
  useTranslation: jest.fn(),
}));

// Mock the Carbon components to avoid dropdown itemToString issues with invalid data
jest.mock('@carbon/react', () => {
  const actual = jest.requireActual('@carbon/react');

  // Define types for the props to avoid TypeScript errors
  interface MockDropdownProps {
    id: string;
    titleText: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    items: Array<any>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    itemToString: (item: any) => string;
    disabled?: boolean;
  }

  return {
    ...actual,
    Dropdown: ({
      id,
      titleText,
      items,
      itemToString,
      disabled,
    }: MockDropdownProps) => {
      // A safer version that prevents edge case errors
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const safeItemToString = (item: any): string => {
        try {
          return itemToString(item);
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
        } catch (e) {
          return '';
        }
      };

      return (
        <div data-testid={id}>
          <div>{titleText}</div>
          <select disabled={disabled} aria-label={titleText}>
            {
              // eslint-disable-next-line @typescript-eslint/no-explicit-any
              items.map((item: any, i: number) => (
                <option
                  key={i}
                  value={typeof item === 'object' && item?.uuid ? item.uuid : i}
                >
                  {safeItemToString(item)}
                </option>
              ))
            }
          </select>
        </div>
      );
    },
  };
});

// Configure jest-axe
expect.extend(toHaveNoViolations);

describe('BasicForm', () => {
  // Common setup
  const mockTranslation = {
    t: jest.fn((key) => key),
  };

  // Updated mock to match actual Provider type structure
  const mockPractitioner: Provider = {
    uuid: 'provider-uuid-123',
    display: 'Dr. Smith - Clinician',
    person: {
      uuid: 'person-uuid-456',
      display: 'Dr. John Smith',
      gender: 'M',
      age: 35,
      birthdate: '1987-01-01T00:00:00.000+0000',
      birthdateEstimated: false,
      dead: false,
      deathDate: null,
      causeOfDeath: null,
      preferredName: {
        uuid: 'name-uuid-789',
        display: 'Dr. John Smith',
        links: [],
      },
      preferredAddress: null,
      attributes: [],
      voided: false,
      birthtime: null,
      deathdateEstimated: false,
      links: [],
      resourceVersion: '1.9',
    },
  };

  const mockVisitTypes = [
    { uuid: '345', name: 'Visit Type 1' },
    { uuid: '678', name: 'Visit Type 2' },
  ];

  const mockEncounterTypes = [
    { uuid: '789', name: 'Encounter Type 1' },
    { uuid: '012', name: 'Encounter Type 2' },
  ];

  const mockLocation = {
    uuid: '123',
    display: 'Location 1',
    links: [], // Added links property to match OpenMRSLocation interface
  };
  const defaultDate = '16/05/2025';

  beforeEach(() => {
    jest.clearAllMocks();
    // Setup default mock implementations
    (useTranslation as jest.Mock).mockReturnValue(mockTranslation);
  });

  const renderBasicForm = (props = {}) => {
    const defaultProps = {
      practitioner: mockPractitioner,
      visitTypes: mockVisitTypes,
      visitTypeSelected: mockVisitTypes[0],
      encounterTypes: mockEncounterTypes,
      encounterTypeSelected: mockEncounterTypes[0],
      location: mockLocation,
      locationSelected: mockLocation,
      defaultDate,
    };

    return render(<BasicForm {...defaultProps} {...props} />);
  };

  describe('Rendering', () => {
    it('should match snapshot', () => {
      // Act
      const { container } = renderBasicForm();

      // Assert
      expect(container).toMatchSnapshot();
    });

    it('should render all form fields with correct data', () => {
      // Act
      renderBasicForm();

      // Assert
      expect(screen.getByText('LOCATION')).toBeInTheDocument();
      expect(screen.getByText('ENCOUNTER_TYPE')).toBeInTheDocument();
      expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
      expect(screen.getByText('ENCOUNTER_DATE')).toBeInTheDocument();
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
    });

    it('should render all form fields as disabled', () => {
      // Act
      renderBasicForm();

      // Assert
      // Use role selectors to find the elements by their roles and accessible names
      const locationDropdown = screen.getByRole('combobox', {
        name: /LOCATION/i,
      });
      const encounterTypeDropdown = screen.getByRole('combobox', {
        name: /ENCOUNTER_TYPE/i,
      });
      const visitTypeDropdown = screen.getByRole('combobox', {
        name: /VISIT_TYPE/i,
      });
      const datePickerInput = screen.getByLabelText(/ENCOUNTER_DATE/i);
      const practitionerDropdown = screen.getByRole('combobox', {
        name: /PARTICIPANT/i,
      });

      expect(locationDropdown).toBeDisabled();
      expect(encounterTypeDropdown).toBeDisabled();
      expect(visitTypeDropdown).toBeDisabled();
      expect(datePickerInput).toBeDisabled();
      expect(practitionerDropdown).toBeDisabled();
    });

    it('should not render Visit Type dropdown when visitTypeSelected is not provided', () => {
      // Act
      renderBasicForm({ visitTypeSelected: null });

      // Assert
      expect(screen.queryByText('VISIT_TYPE')).not.toBeInTheDocument();
    });

    it('should not render Practitioner dropdown when practitioner is not provided', () => {
      // Act
      renderBasicForm({ practitioner: null });

      // Assert
      expect(screen.queryByText('PARTICIPANT')).not.toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('should not have any accessibility violations', async () => {
      // Act
      const { container } = renderBasicForm();

      // Assert
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('should have appropriate ARIA attributes on all form elements', () => {
      // Act
      renderBasicForm();

      // Assert
      // Check that all interactive elements have accessible properties
      expect(
        screen.getByRole('combobox', { name: /LOCATION/i }),
      ).toHaveAttribute('disabled');
      expect(
        screen.getByRole('combobox', { name: /ENCOUNTER_TYPE/i }),
      ).toHaveAttribute('disabled');
      expect(
        screen.getByRole('combobox', { name: /VISIT_TYPE/i }),
      ).toHaveAttribute('disabled');
      expect(screen.getByLabelText(/ENCOUNTER_DATE/i)).toHaveAttribute(
        'disabled',
      );
    });
  });

  describe('Edge Cases', () => {
    it('should handle empty arrays for dropdowns', () => {
      // Act
      renderBasicForm({
        visitTypes: [],
        encounterTypes: [],
      });

      // Assert
      expect(
        screen.getByRole('combobox', { name: /ENCOUNTER_TYPE/i }),
      ).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /VISIT_TYPE/i }),
      ).toBeInTheDocument();
    });

    it('should handle undefined encounterTypes by using an empty array', () => {
      // Act
      renderBasicForm({
        encounterTypes: undefined,
      });

      // Assert
      expect(screen.getByText('ENCOUNTER_TYPE')).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /ENCOUNTER_TYPE/i }),
      ).toBeInTheDocument();
    });

    it('should handle undefined item in visit type itemToString', () => {
      // We need to modify our mock component to test the itemToString function
      // We'll use our mock dropdown to test this by checking it renders even with bad data
      renderBasicForm({
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        visitTypes: [undefined, null, { name: 'Valid Visit Type' }] as any,
      });

      // Assert dropdown still renders
      expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /VISIT_TYPE/i }),
      ).toBeInTheDocument();
    });

    it('should handle undefined location', () => {
      // Act
      renderBasicForm({
        location: undefined,
        locationSelected: undefined,
      });

      // Assert
      expect(
        screen.getByRole('combobox', { name: /LOCATION/i }),
      ).toBeInTheDocument();
    });

    // New test for Provider structure handling
    it('should handle Provider without person property', () => {
      // Act - Using our mock Carbon component to avoid the real error
      renderBasicForm({
        practitioner: { uuid: 'provider-uuid', display: 'Provider Display' },
      });

      // Assert
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
      // With our mock, we can still verify the dropdown is there
      expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
    });

    // New test for Provider structure handling
    it('should handle Provider with person missing preferredName', () => {
      // Act - Using our mock Carbon component to avoid the real error
      const incompleteProvider = {
        ...mockPractitioner,
        person: {
          ...mockPractitioner.person,
          preferredName: undefined,
        },
      };

      renderBasicForm({
        practitioner: incompleteProvider,
      });

      // Assert
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
      // With our mock, we can still verify the dropdown is there
      expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
    });
  });

  describe('Conditional Rendering', () => {
    describe('Visit Type Dropdown', () => {
      it('should render Visit Type dropdown when visitTypeSelected is provided', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: mockVisitTypes[0],
        });

        // Assert
        expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
        const visitTypeDropdown = screen.getByRole('combobox', {
          name: /VISIT_TYPE/i,
        });
        expect(visitTypeDropdown).toBeInTheDocument();
        expect(visitTypeDropdown).toBeDisabled();
      });

      it('should not render Visit Type dropdown when visitTypeSelected is null', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: null,
        });

        // Assert
        expect(screen.queryByText('VISIT_TYPE')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /VISIT_TYPE/i }),
        ).not.toBeInTheDocument();
      });

      it('should not render Visit Type dropdown when visitTypeSelected is undefined', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: undefined,
        });

        // Assert
        expect(screen.queryByText('VISIT_TYPE')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /VISIT_TYPE/i }),
        ).not.toBeInTheDocument();
      });

      it('should render Visit Type dropdown when visitTypeSelected is an empty object', () => {
        // Act
        renderBasicForm({
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          visitTypeSelected: {} as any, // Empty object cast as Concept
        });

        // Assert
        expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
        const visitTypeDropdown = screen.getByRole('combobox', {
          name: /VISIT_TYPE/i,
        });
        expect(visitTypeDropdown).toBeInTheDocument();
      });
    });

    describe('Practitioner Dropdown', () => {
      it('should render Practitioner dropdown when practitioner is provided', () => {
        // Act
        renderBasicForm({
          practitioner: mockPractitioner,
        });

        // Assert
        expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
        const practitionerDropdown = screen.getByRole('combobox', {
          name: /PARTICIPANT/i,
        });
        expect(practitionerDropdown).toBeInTheDocument();
        expect(practitionerDropdown).toBeDisabled();
      });

      it('should not render Practitioner dropdown when practitioner is null', () => {
        // Act
        renderBasicForm({
          practitioner: null,
        });

        // Assert
        expect(screen.queryByText('PARTICIPANT')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /PARTICIPANT/i }),
        ).not.toBeInTheDocument();
      });

      it('should not render Practitioner dropdown when practitioner is undefined', () => {
        // Act
        renderBasicForm({
          practitioner: undefined,
        });

        // Assert
        expect(screen.queryByText('PARTICIPANT')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /PARTICIPANT/i }),
        ).not.toBeInTheDocument();
      });

      it('should render Practitioner dropdown when practitioner is an empty object', () => {
        // Act - Using our mock Carbon component to avoid the real error
        renderBasicForm({
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          practitioner: {} as any, // Empty object cast as Provider
        });

        // Assert
        expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
        // With our mock, we can still verify the dropdown is there
        expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
      });
    });

    describe('Combined Conditional Rendering', () => {
      it('should render both conditionally rendered components when both conditions are met', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: mockVisitTypes[0],
          practitioner: mockPractitioner,
        });

        // Assert
        // Visit Type should be rendered
        expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
        expect(
          screen.getByRole('combobox', { name: /VISIT_TYPE/i }),
        ).toBeInTheDocument();

        // Practitioner should be rendered
        expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
        expect(
          screen.getByRole('combobox', { name: /PARTICIPANT/i }),
        ).toBeInTheDocument();
      });

      it('should not render either component when both conditions are not met', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: null,
          practitioner: null,
        });

        // Assert
        // Visit Type should not be rendered
        expect(screen.queryByText('VISIT_TYPE')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /VISIT_TYPE/i }),
        ).not.toBeInTheDocument();

        // Practitioner should not be rendered
        expect(screen.queryByText('PARTICIPANT')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /PARTICIPANT/i }),
        ).not.toBeInTheDocument();
      });

      it('should render only Visit Type when only visitTypeSelected is provided', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: mockVisitTypes[0],
          practitioner: null,
        });

        // Assert
        // Visit Type should be rendered
        expect(screen.getByText('VISIT_TYPE')).toBeInTheDocument();
        expect(
          screen.getByRole('combobox', { name: /VISIT_TYPE/i }),
        ).toBeInTheDocument();

        // Practitioner should not be rendered
        expect(screen.queryByText('PARTICIPANT')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /PARTICIPANT/i }),
        ).not.toBeInTheDocument();
      });

      it('should render only Practitioner when only practitioner is provided', () => {
        // Act
        renderBasicForm({
          visitTypeSelected: null,
          practitioner: mockPractitioner,
        });

        // Assert
        // Visit Type should not be rendered
        expect(screen.queryByText('VISIT_TYPE')).not.toBeInTheDocument();
        expect(
          screen.queryByRole('combobox', { name: /VISIT_TYPE/i }),
        ).not.toBeInTheDocument();

        // Practitioner should be rendered
        expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
        expect(
          screen.getByRole('combobox', { name: /PARTICIPANT/i }),
        ).toBeInTheDocument();
      });
    });
  });

  // Updated Provider Display Logic test section
  describe('Provider Display Logic', () => {
    it('should show the practitioner dropdown when valid data is provided', () => {
      // Act
      renderBasicForm();

      // Assert - check that the dropdown exists
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
      expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
    });

    // Test for the itemToString function used in the practitioner dropdown
    it('should correctly handle item display logic in encounter type dropdown', () => {
      renderBasicForm({
        encounterTypes: [
          {
            uuid: 123,
          },
        ],
      });
      const encounterTypeDropdown = screen.getByTestId(
        'encounter-type-dropdown',
      );
      const options = encounterTypeDropdown.querySelector('option');
      expect(options?.textContent).toBe('');
    });

    it('should handle practitioner with null preferredName.display value', () => {
      // Create a practitioner with null display in preferredName
      const modifiedPractitioner = {
        ...mockPractitioner,
        person: {
          ...mockPractitioner.person,
          preferredName: {
            ...mockPractitioner.person.preferredName,
            display: null, // Set display to null to trigger the alternate branch
          },
        },
      };

      // Act
      renderBasicForm({
        practitioner: modifiedPractitioner,
      });

      // Assert
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
      expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
    });

    it('should handle practitioner with missing display properties gracefully', () => {
      // Create a practitioner with missing person display property
      const incompleteProvider = {
        ...mockPractitioner,
        person: {
          ...mockPractitioner.person,
          display: undefined,
        },
      };

      // Act - Using our mock Carbon component to avoid the real error
      renderBasicForm({
        practitioner: incompleteProvider,
      });

      // Assert
      expect(screen.getByText('PARTICIPANT')).toBeInTheDocument();
      expect(screen.getByTestId('practitioner-dropdown')).toBeInTheDocument();
    });
  });
});
