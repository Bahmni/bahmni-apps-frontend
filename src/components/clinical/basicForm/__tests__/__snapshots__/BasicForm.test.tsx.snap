// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicForm Rendering should match snapshot 1`] = `
<div>
  <div
    class="cds--css-grid"
  >
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-5 cds--css-grid-column"
      xl="12"
    >
      <div
        data-testid="location-dropdown"
      >
        <div>
          LOCATION
        </div>
        <select
          aria-label="LOCATION"
          disabled=""
        >
          <option
            value="123"
          >
            Location 1
          </option>
        </select>
      </div>
    </div>
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-5 cds--css-grid-column"
      xl="12"
    >
      <div
        data-testid="encounter-type-dropdown"
      >
        <div>
          ENCOUNTER_TYPE
        </div>
        <select
          aria-label="ENCOUNTER_TYPE"
          disabled=""
        >
          <option
            value="789"
          >
            Encounter Type 1
          </option>
          <option
            value="012"
          >
            Encounter Type 2
          </option>
        </select>
      </div>
    </div>
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-5 cds--css-grid-column"
      xl="12"
    >
      <div
        data-testid="visit-type-dropdown"
      >
        <div>
          VISIT_TYPE
        </div>
        <select
          aria-label="VISIT_TYPE"
          disabled=""
        >
          <option
            value="345"
          >
            Visit Type 1
          </option>
          <option
            value="678"
          >
            Visit Type 2
          </option>
        </select>
      </div>
    </div>
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-5 cds--css-grid-column"
    >
      <div
        data-testid="practitioner-dropdown"
      >
        <div>
          PARTICIPANT
        </div>
        <select
          aria-label="PARTICIPANT"
          disabled=""
        >
          <option
            value="provider-uuid-123"
          >
            Dr. John Smith
          </option>
        </select>
      </div>
    </div>
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-5 cds--css-grid-column"
    >
      <div
        class="cds--form-item"
      >
        <div
          class="cds--date-picker cds--date-picker--single"
        >
          <div
            class="cds--date-picker-container"
          >
            <label
              class="cds--label cds--label--disabled"
              dir="auto"
              for="encounter-date-picker-input"
            >
              ENCOUNTER_DATE
            </label>
            <div
              class="cds--date-picker-input__wrapper"
            >
              <span>
                <input
                  class="cds--date-picker__input cds--date-picker__input--md flatpickr-input"
                  disabled=""
                  id="encounter-date-picker-input"
                  pattern="\\d{1,2}\\/\\d{1,2}\\/\\d{4}"
                  placeholder="dd/mm/yyyy"
                  title="ENCOUNTER_DATE"
                  type="text"
                  value="16/05/2025"
                />
                <svg
                  aria-hidden="true"
                  class="cds--date-picker__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20 V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <li
    class="cds--menu-item-divider"
    role="separator"
  />
</div>
`;
